plugins/processors/gpu/processor.go

```diff
@@ -0,0 +1,194 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package gpu
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/zap"
+)
+
+const (
+	gpuMetric                = "_gpu_"
+	gpuContainerMetricPrefix = "container_"
+	gpuPodMetricPrefix       = "pod_"
+	gpuNodeMetricPrefix      = "node_"
+)
+
+var podContainerMetricLabels = map[string]map[string]interface{}{
+	"ClusterName":  nil,
+	"FullPodName":  nil,
+	"PodName":      nil,
+	"InstanceId":   nil,
+	"InstanceType": nil,
+	"NodeName":     nil,
+	"Timestamp":    nil,
+	"Type":         nil,
+	"Version":      nil,
+	"Namespace":    nil,
+	"Sources":      nil,
+	"UUID":         nil,
+	"kubernetes":   nil,
+}
+
+var nodeMetricLabels = map[string]map[string]interface{}{
+	"ClusterName":  nil,
+	"InstanceId":   nil,
+	"InstanceType": nil,
+	"NodeName":     nil,
+	"Timestamp":    nil,
+	"Type":         nil,
+	"Version":      nil,
+	"kubernetes": {
+		"host": nil,
+	},
+}
+
+type gpuprocessor struct {
+	*Config
+	logger     *zap.Logger
+	cancelFunc context.CancelFunc
+	shutdownC  chan bool
```

Comment:
This cancel func doesn't do anything. It's not attached to a context. The `shutdownC` isn't used for anything either. It's just created and then closed. Nothing is blocking on it or listening on the channel.

Reply:
Thanks, start/shutdown are removed.